checker/tests/nullness/FlowField.java

```diff
@@ -33,6 +33,7 @@ void test1() {
         protected String field = null;
     }
 
+    // :: error: (extends.clause.invalid)
     class B extends A {
```

Comment:
Could you explain the motivation for this error? I'm confused why we want it.
Per #2247 we want class declarations to be the top type. What default applies to the extends clause?
This particular example is in a `DefaultQualifier(Nullable.class)` scope, so isn't it
`@Nullable class B extends @Nullable A`?
That would confirm to both the bound from the declaration of A (which is also `@Nullable class A`) and the use as bound for B.

Reply:
I agree that this declaration seems like `@Nullable class B extends @Nullable A`.  Good catch!

#2247 doesn't say all declarations are top type (except in a comment that refers to a proposal for a design change).  We could make such a change in the future.  In the meanwhile, it will be good to make the implementation consistent with the current design.

Reply:
> I agree that this declaration seems like `@Nullable class B extends @Nullable A`. Good catch!

I did a little debugging and found that the nullness checker replaces the annotation on `class A` with `@NonNull`. This replacement is done in the `NullnessAnnotatedTypeFactory`:
https://github.com/typetools/checker-framework/blob/513dda9710ba76e7d6d742d08b5972fdd7f7605b/checker/src/main/java/org/checkerframework/checker/nullness/NullnessAnnotatedTypeFactory.java#L379

Reply:
That method is ugly.  Here is a pull request that eliminates it: https://github.com/typetools/checker-framework/pull/2292 .